### ISSUE TITLE
Bug: parse error on valid code in eslint 8

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#15234]
### Environment

Node version: 17.0.1
npm version: 8.1.0
Local ESLint version: 8.1.0
Global ESLint version: n/a
Operating System: Mac

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>
Using RuleTester, so virtually none:

```json
{
  "ecmaVersion": 2022,
  "sourceType": "module",
  "ecmaFeatures": {
    "jsx": true,
  },
}
```
</details>

```jsx
        class Foo extends React.Component {
          contructor(props) {
            super(props);
            this.initialValues = {
              test: '',
            };
          }

          render = () => {
            return (
              <Component
                initialValues={this.props.initialValues || this.initialValues}
              >
                {formikProps => (
                  <Input {...formikProps} />
                )}
              </Component>
            );
          }
        }
```

### What did you expect to happen?

A successful parse.

### What actually happened?

`A fatal parsing error occurred: Parsing error: super() call outside constructor of a subclass`

This is a `super()` call inside the constructor of a subclass.

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

Happy to send a PR but I have no idea where to send it.

**Comment by mdjermanovic** [comment]
> ```js
>           contructor(props) {
> ```

It's a typo, missing "s": `contructor` -> `constructor`

**Comment by mdjermanovic** [comment]
Closing, as this works as intended. Parser correctly reports an error on use of `super()` call in a method.

When the typo is fixed and the method becomes the constructor, there are no parsing errors, as in [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiICAgICAgICBjbGFzcyBGb28gZXh0ZW5kcyBSZWFjdC5Db21wb25lbnQge1xuICAgICAgICAgIGNvbnN0cnVjdG9yKHByb3BzKSB7XG4gICAgICAgICAgICBzdXBlcihwcm9wcyk7XG4gICAgICAgICAgICB0aGlzLmluaXRpYWxWYWx1ZXMgPSB7XG4gICAgICAgICAgICAgIHRlc3Q6ICcnLFxuICAgICAgICAgICAgfTtcbiAgICAgICAgICB9XG5cbiAgICAgICAgICByZW5kZXIgPSAoKSA9PiB7XG4gICAgICAgICAgICByZXR1cm4gKFxuICAgICAgICAgICAgICA8Q29tcG9uZW50XG4gICAgICAgICAgICAgICAgaW5pdGlhbFZhbHVlcz17dGhpcy5wcm9wcy5pbml0aWFsVmFsdWVzIHx8IHRoaXMuaW5pdGlhbFZhbHVlc31cbiAgICAgICAgICAgICAgPlxuICAgICAgICAgICAgICAgIHtmb3JtaWtQcm9wcyA9PiAoXG4gICAgICAgICAgICAgICAgICA8SW5wdXQgey4uLmZvcm1pa1Byb3BzfSAvPlxuICAgICAgICAgICAgICAgICl9XG4gICAgICAgICAgICAgIDwvQ29tcG9uZW50PlxuICAgICAgICAgICAgKTtcbiAgICAgICAgICB9XG4gICAgICAgIH0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEzLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6e319fQ==).

**Comment by ljharb** [comment]
Wow, ok thanks - in that case it’s a bug in a few other parsers that accept the typo’d version :-)
